backend/ringv2/ringv2.go

```diff
@@ -355,12 +363,22 @@ func (r *Ring) startWatchers(ctx context.Context, ch chan Event, name string, va
 	r.mu.Lock()
 	r.watchers[watcher.watcherKey] = watcher
 	r.mu.Unlock()
-	if err := watcher.ensureActiveTrigger(ctx); err != nil {
-		notifyError(ch, fmt.Errorf("error while starting ring watcher: %s", err))
-		notifyClosing(ch)
-		cancel()
-		return
+
+	backoff := retry.ExponentialBackoff{
+		InitialDelayInterval: 10 * time.Millisecond,
+		MaxDelayInterval:     10 * time.Second,
+		Multiplier:           10,
+		Ctx:                  ctx,
 	}
+
+	_ = backoff.Retry(func(retry int) (bool, error) {
+		if err := watcher.ensureActiveTrigger(ctx); err != nil {
+			notifyError(ch, fmt.Errorf("error while starting ring watcher: %s", err))
```

Comment:
@echlebek while the bug is fixed with this change, I'm curious if I still need to notify the closing channel and cancel:
```suggestion
			notifyError(ch, fmt.Errorf("error while starting ring watcher: %s", err))
			notifyClosing(ch)
			cancel()
```

Reply:
I think you should capture the error that comes from `backoff.Retry` and cancel if it's non-nil.